shaping/wrapping.go

```diff
@@ -362,6 +363,200 @@ func (r *runMapper) mapRun(runIdx int, run Output) {
 	}
 }
 
+// RunIterator defines a type that can incrementally provide shaped text.
+type RunIterator interface {
+	// Next returns the next logical run index, shaped run, and true if there are more runs,
+	// then it advances the iterator to the next element.
+	// Otherwise it returns an undefined index, an empty Output, and false.
+	Next() (index int, run Output, isValid bool)
+	// Peek returns the same thing Next() would, but does not advance the iterator (so the
+	// next call to Next() will return the same thing).
+	Peek() (index int, run Output, isValid bool)
+	// Save marks the current iterator position such that the iterator can return to it later
+	// when Restore() is called. Only one position may be saved at a time, with subsequent
+	// calls to Save() overriding the current value.
+	Save()
+	// Restore resets the iteration state to the most recently Save()-ed position.
+	Restore()
+}
+
+// runSlice is a [RunIterator] built from already-shaped text.
+type runSlice struct {
```

Comment:
wouldn't 'shapedRun' be more descriptive?

Reply:
As in `shapedRunSlice`? Sure